src/display-buffer.coffee

```diff
@@ -332,6 +322,14 @@ class DisplayBuffer extends Model
     @scopedCharacterWidthsChangeCount++
     @characterWidthsChanged() unless @batchingCharacterMeasurement
 
+  setCharWidthsForLineId: (lineId, charWidths) ->
+    @charWidthsForLineId[lineId] = charWidths
+    @characterWidthsChanged() unless @batchingCharacterMeasurement
+
+  getCharWidthForLine: (line, index) ->
```

Comment:
I wonder if this should just be `getCharacterWidth` and take a `Point`. It could then consult its lines internally.